src/coreclr/md/compiler/regmeta.cpp

```diff
@@ -246,7 +247,7 @@ RegMeta::CreateNewMD()
     ModuleRec *pModule;
     GUID       mvid;
     IfFailGo(m_pStgdb->m_MiniMd.AddModuleRecord(&pModule, &iRecord));
-    IfFailGo(CoCreateGuid(&mvid));
+    IfFailGo(minipal_guid_v4_create(reinterpret_cast<minipal_guid_t*>(&mvid)) ? S_OK : E_FAIL);
```

Comment:
We should create a `minipal_guid_v4_create<T>(T*)` in the minipal where `sizeof(T) == sizeof(minipal_guid_t)` so we can avoid the casting.

```c++
#ifdef __cplusplus
template<typename T>
bool minipal_guid_v4_create(T* d)
{
    static_assert(sizeof(T) == sizeof(minipal_guid_t));
    return minipal_guid_v4_create(reinterpret_cast<minipal_guid_t*>(d));
}
#endif //__cplusplus
```